src/test/java/com/amazon/opendistroforelasticsearch/security/ccstest/CrossClusterSearchTests.java

```diff
@@ -827,4 +879,113 @@ public void testCcsAggregationsDnfof() throws Exception {
+                .source("{\"cluster\": \""+cl1Info.clustername+"\"}", XContentType.JSON)).actionGet();
+        }
+
+        try (TransportClient tc = getInternalTransportClient(cl2Info, cluster2.transportClientSettings())) {
+            tc.index(new IndexRequest("twitter").type("tweet").setRefreshPolicy(RefreshPolicy.IMMEDIATE).id("0")
+                .source("{\"cluster\": \""+cl2Info.clustername+"\"}", XContentType.JSON)).actionGet();
+        }
+    }
+
+    @Test
+    public void testCcsWithDiffCertsWithNoNodesDnUpdate() throws Exception {
+        final ClusterTransportClientSettings cluster1 = new ClusterTransportClientSettings();
+        final ClusterTransportClientSettings cluster2 = getBaseSettingsWithDifferentCert();
+
+        setupCcs(new DynamicSecurityConfig(), cluster1, cluster2);
+        populateBaseData(cluster1, cluster2);
+
+        String uri = "cross_cluster_two:twitter/tweet/_search?pretty";
+        HttpResponse ccs = rh1.executeGetRequest(uri, encodeBasicHeader("twitter", "nagilum"));
+        System.out.println(ccs.getBody());
+        assertThat(ccs.getStatusCode(), equalTo(HttpStatus.SC_INTERNAL_SERVER_ERROR));
+        assertThat(ccs.getBody(), containsString("no OID or security.nodes_dn incorrect configured"));
+    }
+
+    @Test
+    public void testCcsWithDiffCertsWithNodesDnStaticallyAdded() throws Exception {
+        final ClusterTransportClientSettings cluster1 = new ClusterTransportClientSettings();
+        ClusterTransportClientSettings cluster2 = getBaseSettingsWithDifferentCert();
+        Settings updatedCluster2 = Settings.builder()
+            .put(cluster2.clusterSettings())
+            .putList(ConfigConstants.OPENDISTRO_SECURITY_NODES_DN,
+                "EMAILADDRESS=unt@tst.com,CN=node-untspec5.example.com,OU=SSL,O=Te\\, st,L=Test,C=DE",
+                "CN=node-0.example.com,OU=SSL,O=Test,L=Test,C=DE")
+            .build();
+        cluster2 = new ClusterTransportClientSettings(updatedCluster2, cluster2.transportClientSettings());
+
+        setupCcs(new DynamicSecurityConfig(), cluster1, cluster2);
+        populateBaseData(cluster1, cluster2);
+
+        String uri = "cross_cluster_two:twitter/tweet/_search?pretty";
+        HttpResponse ccs = rh1.executeGetRequest(uri, encodeBasicHeader("twitter", "nagilum"));
+        System.out.println(ccs.getBody());
+        assertThat(ccs.getStatusCode(), equalTo(HttpStatus.SC_OK));
+        assertThat(ccs.getBody(), not(containsString("security_exception")));
+        assertThat(ccs.getBody(), containsString("\"timed_out\" : false"));
+        assertThat(ccs.getBody(), not(containsString("crl1")));
+        assertThat(ccs.getBody(), containsString("crl2"));
+        assertThat(ccs.getBody(), containsString("cross_cluster_two:twitter"));
+    }
+
+    @Test
+    public void testCcsWithDiffCertsWithNodesDnDynamicallyAdded() throws Exception {
```

Comment:
should we also add -ve test? where we do not add new cert dynamically and statically?

Reply:
There is, testCcsWithDiffCertsWithNoNodesDnUpdate()